scripts/quests/windurst/To_Catch_A_Falling_Star.lua

```diff
@@ -0,0 +1,151 @@
+-- Sigismund : !pos -110 -9 80 206
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require("scripts/globals/items")
+require("scripts/globals/npc_util")
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+
+local westSaruIDs = require("scripts/zones/West_Sarutabaruta/IDs")
+-----------------------------------
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.TO_CATCH_A_FALLING_STAR)
+
+quest.reward =
+{
+    fame = 75,
+    fameArea = xi.quest.fame_area.WINDURST,
+    item = xi.items.FISH_SCALE_SHIELD
+}
+
+quest.sections =
+{
+    -- Quest Acceptance
+    {
+        check = function(player, status, vars)
+           return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Sigismund'] = quest:progressEvent(196, 0, xi.items.STARFALL_TEAR),
+
+            onEventFinish =
+            {
+                [196] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    -- Supporting Character dialogue after Accepted
+    {
+        check  = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Tohopka'] = quest:progressEvent(198, 0, xi.items.STARFALL_TEAR, xi.items.HANDFUL_OF_PUGIL_SCALES)
+        }
+    },
+
+    -- Twinkle Tree in West Saru.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WEST_SARUTABARUTA] =
+        {
+            ['Twinkle_Tree'] =
+            {
+                onTrigger = function(player, npc)
+                    if VanadielHour() <= 3 then
+                        player:messageSpecial(westSaruIDs.text.FROST_DEPOSIT_TWINKLES)
+                        return quest:messageSpecial(westSaruIDs.text.MELT_BARE_HANDS)
+                    end
+                    return quest:messageSpecial(westSaruIDs.text.NOTHING_OUT_OF_ORDINARY)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if VanadielHour() > 3 then
+                        return quest:messageSpecial(westSaruIDs.text.NOTHING_OUT_OF_ORDINARY)
+                    end
+
+                    if npcUtil.tradeHas(trade, xi.items.HANDFUL_OF_PUGIL_SCALES) then
+                        if npcUtil.giveItem(player, xi.items.STARFALL_TEAR) then
+                            player:confirmTrade()
+                            if quest:getVar(player, 'Prog') == 0 then
+                                quest:setVar(player, 'Prog', 1)
+                            end
+                            player:messageSpecial(westSaruIDs.text.FROST_DEPOSIT_TWINKLES)
+                            return quest:messageSpecial(westSaruIDs.text.MELT_BARE_HANDS)
+                        end
+                    end
+                    return quest:noAction()
```

Comment:
Shouldn't need a noAction() here.

Reply:
Should return just the quest object itself?  I figured noAction() to enforce the fact that nothing should happen.

Reply:
if nothing else happens, then no code is necessary here.

Reply:
but an onTrigger/onTrade has to return some form of the quest object, no?

Reply:
not if it's doing nothing. if it's outputting text, or w/e and has a purpose it needs to return a quest object. If nothing occurs, then having nothing is okay :)